src/org/zaproxy/zap/extension/pscanrulesBeta/CookieLooselyScopedScanner.java

```diff
@@ -83,12 +83,12 @@ private boolean isLooselyScopedCookie(HttpCookie cookie, String host) {
 		// preconditions
 		assert cookie != null;
 		assert host != null;
-		
+
 		String cookieDomain = cookie.getDomain();
-		
+
 		// if Domain attribute hasn't been specified, the cookie
 		// is scoped with the response host
-		if (cookieDomain == null) {
+		if (cookieDomain == null || cookieDomain.equalsIgnoreCase("null") || cookieDomain.isEmpty()) {

```

Comment:
Well, having the domain set to "null" should not be treated differently than any other value. For the latter case, the domain is only "null" if the request header does not have a target host (which should have).